st2common/st2common/models/api/keyvalue.py

```diff
@@ -57,8 +76,39 @@ class KeyValuePairAPI(BaseAPI):
         'additionalProperties': False
     }
 
+    @staticmethod
+    def _setup_crypto():
+        LOG.info('Checking if encryption is enabled for key-value store.')
+        KeyValuePairAPI.is_encryption_enabled = cfg.CONF.keyvalue.enable_encryption
+        LOG.debug('Encryption enabled? : %s', KeyValuePairAPI.is_encryption_enabled)
+        if KeyValuePairAPI.is_encryption_enabled:
+            KeyValuePairAPI.crypto_key_path = cfg.CONF.keyvalue.encryption_key_path
+            LOG.info('Encryption enabled. Looking for key in path %s',
+                     KeyValuePairAPI.crypto_key_path)
+            if not os.path.exists(KeyValuePairAPI.crypto_key_path):
+                msg = ('Encryption key file does not exist in path %s.' %
+                       KeyValuePairAPI.crypto_key_path)
+                LOG.exception(msg)
+                LOG.info('All API requests will now send out BAD_REQUEST ' +
+                         'if you ask to store secrets in key value store.')
+                KeyValuePairAPI.crypto_key = None
+            else:
+                KeyValuePairAPI.crypto_key = KeyValuePairAPI._read_crypto_key(
```

Comment:
Yeah, I went both ways. Initially, I was loading it on every call to encrypt/decrypt but that definitely seems excessive. I do agree with your assessment of not holding onto the key in memory but then if we load it every time, people could swap out the key and not realize they changed it. If that's beyond us, I can change this to load on every encrypt/decrypt call.